src/main/java/de/medizininformatik_initiative/process/data_transfer/service/DecryptData.java

```diff
@@ -84,17 +84,16 @@ protected void doExecute(DelegateExecution execution, Variables variables)
 					ConstantsDataTransfer.CODESYSTEM_DATA_TRANSFER_VALUE_DATA_SET_STATUS, "Decrypt data-set failed"));
 			variables.updateTask(task);
 
-			variables.setString(ConstantsDataTransfer.BPMN_EXECUTION_VARIABLE_DATA_RECEIVE_ERROR_MESSAGE,
-					"Decrypt data-set failed");
-
 			logger.warn(
 					"Could not decrypt data-set with id '{}' from organization '{}' and project-identifier '{}' referenced in Task with id '{}' - {}",
 					variables.getString(ConstantsDataTransfer.BPMN_EXECUTION_VARIABLE_DATA_SET_REFERENCE),
 					task.getRequester().getIdentifier().getValue(),
 					variables.getString(ConstantsDataTransfer.BPMN_EXECUTION_VARIABLE_PROJECT_IDENTIFIER), task.getId(),
 					exception.getMessage());
-			throw new BpmnError(ConstantsDataTransfer.BPMN_EXECUTION_VARIABLE_DATA_RECEIVE_ERROR,
-					"Decrypt data-set - " + exception.getMessage());
+
+			String error = "Decrypt data-set failed - " + exception.getMessage();
+			variables.setString(ConstantsDataTransfer.BPMN_EXECUTION_VARIABLE_DATA_RECEIVE_ERROR_MESSAGE, error);
```

Comment:
Could we use the [camunda:errorMessageVariable](https://docs.camunda.org/manual/7.22/reference/bpmn20/events/error-events/#defining-an-error) extension in the bpmn to remove this line? If i read the documentation correctly this would use the error message from the BpmnError Exception and store it in a variable defined in the bpmn via the "errorMessageVariable" field. Setting the value from `ConstantsDataTransfer.BPMN_EXECUTION_VARIABLE_DATA_RECEIVE_ERROR_MESSAGE` in the bpmn, may make this line obsolete, without needing to change `HandleErrorReceive`.

This would of cause be true for other error events in the send and receive processes.

Reply:
I absolutely agree. With the goal of having error messages defined consistently in MII processes, this is how it is implemented in other processes as well. We will refactor this part of the code when upgrading to the new process API in v2.

Reply:
Created an issue for this: https://github.com/medizininformatik-initiative/mii-process-data-transfer/issues/29